packages/core/src/services/evaluationResults/aggregations/countersQuery.ts

```diff
@@ -0,0 +1,84 @@
+import { and, count, eq, sum } from 'drizzle-orm'
+
+import { getCommitFilter } from '../_createEvaluationResultQuery'
+import {
+  Commit,
+  Evaluation,
+} from '../../../browser'
+import { database } from '../../../client'
+import { EvaluationResultsRepository } from '../../../repositories'
+import { DocumentLogsRepository } from '../../../repositories/documentLogsRepository'
+import { commits, providerLogs } from '../../../schema'
+
+export async function getEvaluationTotalsQuery(
+  {
+    workspaceId,
+    evaluation,
+    documentUuid,
+    commit,
+  }: {
+    workspaceId: number
+    evaluation: Evaluation
+    documentUuid: string
+    commit: Commit
+  },
+  db = database,
+) {
+  const { scope: evaluationResultsScope } = new EvaluationResultsRepository(
+    workspaceId,
+    db,
+  )
+
+  const documentLogsRepo = new DocumentLogsRepository(workspaceId, db)
+  const documentLogsScope = documentLogsRepo.scope
+
+  const result = await db
+    .select({
+      tokens: sum(providerLogs.tokens).mapWith(Number).as('tokens'),
+      costInMillicents: sum(providerLogs.costInMillicents)
+        .mapWith(Number)
+        .as('cost_in_millicents'),
+    })
+    .from(evaluationResultsScope)
+    .innerJoin(
+      documentLogsScope,
+      eq(documentLogsScope.id, evaluationResultsScope.documentLogId),
+    )
+    .innerJoin(
+      providerLogs,
+      eq(providerLogs.id, evaluationResultsScope.providerLogId),
+    )
+    .where(
+      and(
+        eq(evaluationResultsScope.evaluationId, evaluation.id),
+        eq(documentLogsScope.documentUuid, documentUuid),
+      ),
+    )
+
+  const resultCount = await db
+    .select({
+      totalCount: count(evaluationResultsScope.id),
+    })
+    .from(evaluationResultsScope)
+    .innerJoin(
+      documentLogsScope,
+      eq(documentLogsScope.id, evaluationResultsScope.documentLogId),
+    )
+    .innerJoin(commits, eq(commits.id, documentLogsScope.commitId))
+    .where(
+      and(
+        eq(evaluationResultsScope.evaluationId, evaluation.id),
+        eq(documentLogsScope.documentUuid, documentUuid),
+        getCommitFilter(commit),
```

Comment:
Implement filtering by all the versions of the documents since it was changed until the version of the document is in the current commit in the screen. 

I need to talk more offline about this. Not really sure. 

I think this way keeps current behaviour